packages/draw/src/artist.ts

```diff
@@ -221,42 +125,61 @@ export class Artist {
     );
   }
 
-  /**
-    The purpose of this method is to take all of the current drawables and distribute them amongst the layers.
-
-    Currently, there are two layers: an active layer, and a base layer.
-    The base layer is where drawables are typically rendered, unless they are "active" (which is a condition defined by the drawable itself)
-    When a drawable is "active" it is moved to the active layer - currently, this just handles opacity and "focus"
-    If ANY drawables are on the active layer, the base layer is dimmed.
-  */
   private distribute_drawables(): void {
+    const now = new Date().getTime();
+    const activeAtStart = this.active_layer.drawables.length;
+
+    this.active_layer.drawables = [];
+    this.purgatory.drawables = [];
+    this.base_layer.drawables = [];
+
     for (const d of this.drawables) {
-      if (this.cursor && d.isActive(this.cursor, this.zoomer)) {
-        this.transitionManager.transitionToLayerWithAnimation({
-          drawable: d,
-          sourceLayer: this.base_layer,
-          targetLayer: this.active_layer,
-          animationConfig: new Map([
-            [this.base_layer.name, [DIMMING_ANIMATION_CONFIG(this.styles)]],
-          ]),
-          focus: 'active',
-          transitionDuration: 0,
-          transitionName: 'baseToActive',
-        });
+      if (
+        this.cursor &&
+        d.isActive({ x: this.cursor.x, y: this.cursor.y }, this.zoomer) &&
+        !this.active_layer.drawables.includes(d)
+      ) {
+        this.active_layer.drawables.push(d);
+      } else if (
+        typeof d.lastTimeActive === 'number' &&
+        now - d.lastTimeActive <= this.styles.dimmingLayerDuration * 1000
+      ) {
+        this.purgatory.drawables.push(d);
       } else {
-        this.transitionManager.transitionToLayerWithAnimation({
-          drawable: d,
-          sourceLayer: this.active_layer,
-          targetLayer: this.base_layer,
-          animationConfig: new Map([
-            [this.base_layer.name, [BRIGHTENING_ANIMATION_CONFIG(this.styles)]],
-          ]),
-          focus: 'neutral',
-          transitionDuration: this.styles.dimmingLayerDuration,
-          transitionName: 'activeToBase',
-        });
+        this.base_layer.drawables.push(d);
       }
     }
+
+    if (this.active_layer.drawables.length) {
+      this.base_layer.focus = 'inactive';
+      do {
+        const node = this.purgatory.drawables.pop();
+        if (node) {
+          node.reset();
+          this.base_layer.drawables.push(node);
+        }
+      } while (this.purgatory.drawables.length);
```

Comment:
This rips nodes out of purgatory and resets them to base if you make a new group active while purgatory nodes are in the middle of their animation.